### DISCUSSION TITLE
Intermittent Failure Because "Detect Begin" Called after "Apply Complete"

### DISCUSSION THREAD

**Discussion by afbarksd** [wixtoolset/issues#7799] [Questions]
We have an intermittent problem while installing a product using a Chocolatey package that employs WiX on a new virtual machine where the product has never before been installed.  Usually, the installation of the product's two MSI packages works as expected, and the WiX log  file includes the following:

     Apply complete, result: 0x0, restart: None, ba requested restart:  No
     Shutting down, exit code: 0x0

Sometimes, with the same Chocolatey package and installing in the same way on a new virtual machine, the installation fails.  The log file diverges from the log file of a successful installation after the "Apply Complete" line.

     Apply complete, result: 0x0, restart: None, ba requested restart:  No
     Detect begin, 2 packages

In the failed installation, the log file shows that WiX proceeds to find the two MSI packages that it just installed and raises an exception because the two MSI packages that it just installed are indeed installed.  It is as if the installation started over.  As I wrote above, this happens intermittently during the first installation of the product on a new virtual machine.

**Comment by barnson** [comment]
Sounds like a buggy BA.
